packages/widgets/src/blocks/legacy-widget/edit/preview.js

```diff
@@ -7,24 +7,25 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { useRefEffect } from '@wordpress/compose';
-import { addQueryArgs } from '@wordpress/url';
-import { useState, useContext } from '@wordpress/element';
+import { useState } from '@wordpress/element';
 import { Placeholder, Spinner, Disabled } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
 
-/**
- * Internal dependencies
- */
-import WidgetsSettings from '../../../components/widgets-settings';
+async function renderPreview( idBase, instance ) {
```

Comment:
Nit: I would probably go for `getPreviewHTML` or something like that but that's just my preference.